app/scripts/metamask-controller.js

```diff
@@ -88,29 +89,20 @@ module.exports = class MetamaskController extends EventEmitter {
     })
 
     // currency controller
-    this.currencyController = new CurrencyController({
-      initState: initState.CurrencyController,
-    })
-    this.currencyController.updateConversionRate()
-    this.currencyController.scheduleConversionInterval()
+    this.currencyRateController = new CurrencyRateController(initState.CurrencyRateController)
 
-    // infura controller
-    this.infuraController = new InfuraController({
-      initState: initState.InfuraController,
-    })
-    this.infuraController.scheduleInfuraNetworkCheck()
+    // network status controller
+    this.networkStatusController = new NetworkStatusController(initState.NetworkStatusController)
 
-    this.blacklistController = new BlacklistController()
-    this.blacklistController.scheduleUpdates()
+    this.phishingController = new PhishingController()
 
     // rpc provider
     this.provider = this.initializeProvider()
     this.blockTracker = this.provider._blockTracker
 
     // token exchange rate tracker
-    this.tokenRatesController = new TokenRatesController({
-      preferences: this.preferencesController.store,
-    })
+    this.tokenRatesController = new TokenRatesController()
+    this.preferencesController.store.subscribe(({ tokens = [] }) => { this.tokenRatesController.tokens = tokens })
```

Comment:
This is most likely temporary. The `gaba/ComposableController` (which will ultimately be used in the new MetaMaskController) passes a `context` property to each child controller that contains a reference to all other child controllers. This allows child controllers to subscribe to sibling state changes (and call sibling API methods.)

There is some elegance to being explicit about interdependent state updates though. I'm hoping the more pragmatic approach will become evident when we finally get to MetaMaskController.